addons/stock_account/models/product.py

```diff
@@ -60,7 +60,7 @@ def write(self, vals):
             raise UserError(_("The action leads to the creation of a journal entry, for which you don't have the access rights."))
         # Create the account moves.
         if move_vals_list:
-            account_moves = self.env['account.move'].sudo().create(move_vals_list)
+            account_moves = self.env['account.move'].sudo().with_context(default_type='entry').create(move_vals_list)
```

Comment:
It's probably better to complete _svl_replenish_stock_am and _svl_empty_stock_am than using a context key